arai60/21-29_Tree_BT_BST/21_104_Maximum Depth of Binary Tree/level_1.py

```diff
@@ -0,0 +1,47 @@
+# DFS
+class Solution:
+    def maxDepth(self, root: Optional[TreeNode]) -> int:
+
+        def search_max_depth(node):
+            if not node:
+                return 0
+            left_depth = search_max_depth(node.left) + 1
+            right_depth = search_max_depth(node.right) + 1
+            max_depth = max(left_depth, right_depth)
+            return max_depth
+
+        return search_max_depth(root)
+
+
+# BFS
+class Solution:
+    def maxDepth(self, root: Optional[TreeNode]) -> int:
+        if not root:
+            return 0
+        node_depth_queue = deque([(root, 1)])
+        max_depth = 0
+        while node_depth_queue:
+            node, depth = node_depth_queue.popleft()
+            if node.left:
+                node_depth_queue.append((node.left, depth + 1))
+            if node.right:
+                node_depth_queue.append((node.right, depth + 1))
+            max_depth = max(max_depth, depth)
```

Comment:
BFSなのでmax取らなくても良いですね。
```suggestion
            max_depth = depth
```

Reply:
一番最後のdepthが最大値で決まってますね。
max_depth変数作らずにdepthを返しても動きますね。